webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/facts/nodes/NodeFactRepository.scala

```diff
@@ -0,0 +1,241 @@
+/*
+*************************************************************************************
+* Copyright 2021 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+package com.normation.rudder.facts.nodes
+
+import com.normation.errors.IOResult
+import com.normation.inventory.domain._
+import com.normation.rudder.apidata.FullDetailLevel
+import com.normation.rudder.domain.nodes.NodeInfo
+import com.normation.rudder.git.GitItemRepository
+import com.normation.rudder.git.GitRepositoryProvider
+
+import better.files.File
+import net.liftweb.json.prettyRender
+import net.liftweb.json.JsonDSL._
+import com.softwaremill.quicklens._
+import org.eclipse.jgit.lib.PersonIdent
+
+import zio._
+import zio.syntax._
+import com.normation.errors._
+
+/*
+ * write node facts
+ */
+trait NodeFactRepository {
```

Comment:
We may want to do a FactRepository[A,B] that goes well with the SerializeFacts trait

Reply:
at least persist

Reply:
yeah, but here, too, I'm not sure how to do it, b/c I'm not sure that we want to have only one big fact repository with all fact, and thus, it should have one persist by type (and so param is at persist level), or if we want one small repo by fact (like what we have for configuration). The second solution may feel simpler, but it leads to a LOT of duplication of code, and maintenance is not easy. I will add comment, too.